### ISSUE TITLE
Objective-C: Update to grpc 0.15.0 error

### ISSUE THREAD

**Issue by HaloWang** [grpc/grpc#7231]
I install grpc 0.15.0 using homebrew:
`brew install grpc --with-plugins`
In my local `Protos.podspec`, I ask cocoapods to install `grpc-0.14.0`:
```
Pod::Spec.new do |s|
  s.name     = "Protos"
  s.version  = "0.0.1"
  s.license  = "New BSD"
  s.authors  = { 'gRPC contributors' => 'grpc-io@googlegroups.com' }
  s.homepage = "http://www.grpc.io/"
  s.summary  = "Protos"
  s.source   = { :git => 'https://github.com/grpc/grpc.git' }

  s.ios.deployment_target = "7.1"
  s.osx.deployment_target = "10.9"
  s.prepare_command       = "protoc --objc_out=. --objcgrpc_out=. **/*.proto"

  s.subspec "Messages" do |ms|
    ms.source_files        = "*.pbobjc.{h,m}", "**/*.pbobjc.{h,m}"
    ms.header_mappings_dir = "."
    ms.requires_arc        = false
    ms.dependency "Protobuf", "3.0.0-beta-3.1"
  end

  s.subspec "Services" do |ss|
    ss.source_files        = "*.pbrpc.{h,m}", "**/*.pbrpc.{h,m}"
    ss.header_mappings_dir = "."
    ss.requires_arc        = true
    ss.dependency "gRPC", "0.14.0"
    ss.dependency "#{s.name}/Messages"
  end
end
```
In my podfile:
```
target 'My_Target' do
  pod 'Protos', :path=>"./Protos"
end
```
Then I use `pod install`(my pod version is 1.0.1) and success.
But when I open my xcworkspace and build my project, so many same errors:
/Users/WangCe/Desktop/GT3/Pods/Headers/Private/Protos/init/AppInit.pbrpc.h:3:9: 'ProtoRPC/ProtoService.h' file not found
/Users/WangCe/Desktop/GT3/Pods/Headers/Private/Protos/event/BannerEvent.pbrpc.h:3:9: 'ProtoRPC/ProtoService.h' file not found
/Users/WangCe/Desktop/GT3/Pods/Headers/Private/Protos/order/OrderBuyer.pbrpc.h:3:9: 'ProtoRPC/ProtoService.h' file not found
/Users/WangCe/Desktop/GT3/Pods/Headers/Private/Protos/order/OrderDetail.pbrpc.h:3:9: 'ProtoRPC/ProtoService.h' file not found
/Users/WangCe/Desktop/GT3/Pods/Headers/Private/Protos/person/RealnameCertificate.pbrpc.h:3:9: 'ProtoRPC/ProtoService.h' file not found
/Users/WangCe/Desktop/GT3/Pods/Headers/Private/Protos/store/Store.pbrpc.h:3:9: 'ProtoRPC/ProtoService.h' file not found

I clone the latest version of grpc from Github, and install it as a gitsubmodule:
.gitsubmodule:
```
[submodule "grpc"]
	path = grpc
	url = https://github.com/grpc/grpc.git
```
Podfile:
```
   ...
  pod 'Protos', :path=>"./MyProtos"
  pod 'gRPC',      :path => "./grpc"
  pod 'Protobuf',  :path => "./grpc/third_party/protobuf/"
  pod 'BoringSSL', :podspec => "./grpc/src/objective-c"
  ...
```
And build success ✅

Is this the right to use gprc 0.15.0 + ?

**Comment by steeve** [comment]
yeah it seems the Podspec is lagging behind

**Comment by jcanizales** [comment]
Hi guys,

as of PR #7313, if you write your proto-file podspecs [like this](https://github.com/grpc/grpc/blob/master/src/objective-c/README.md#cocoapods) (or like in [this example app](https://github.com/grpc/grpc/blob/master/examples/objective-c/helloworld/HelloWorld.podspec)), the versions of `protoc`, of the gRPC plugin, and of the gRPC runtime will always be resolved by Cocoapods to be compatible.

Specifically, the 0.15 branch lacked some of the fixes of the 0.14 branch for Objective-C, creating this mess. With the new way to use `protoc` via Cocoapods these things won't be a problem ever again.

Please give it a try, forget about 0.15 for Objective-C (our next release after 0.14 will be 1.0, which already had [its branch](https://github.com/grpc/grpc/tree/v1.0.x) cut), and let me know if it works for you!

**Comment by jcanizales** [comment]
This shouldn't be an issue anymore.
